x/dualstaking/keeper/delegate.go

```diff
@@ -561,3 +561,19 @@ func (k Keeper) getUnbondHoldBlocks(ctx sdk.Context, chainID string) uint64 {
 
 	// NOT REACHED
 }
+
+// GetDelegatorProviders gets all the providers the delegator is delegated to
+func (k Keeper) GetDelegatorProviders(ctx sdk.Context, delegator string, epoch uint64) (providers []string, err error) {
+	var delegatorEntry types.Delegator
+	prefix := types.DelegatorKey(delegator)
+
+	found := k.delegatorFS.FindEntry(ctx, prefix, epoch, &delegatorEntry)
+	if !found {
+		return nil, utils.LavaFormatWarning("could not get delegator providers", fmt.Errorf("delegator not found"),
```

Comment:
why is this an error?
I'd think that querying for delegations of a (potential) delegator would return empty list if that (potential) delegator doesn't have any delegations, rather than an error.

Reply:
also, in comparison querying for provider's delegation indeed return empty list if there are none (and not an error)

Reply:
also, the code should validate the input delegator

Reply:
done. see https://github.com/lavanet/lava/commit/d044fe62749f4d88f2bde153789d8316749618bd